lib/c.c

```diff
@@ -325,9 +326,21 @@ void abort()
 FILE *fopen(char *filename, char *mode)
 {
     if (!strcmp(mode, "wb"))
+#ifdef __arm__
         return __syscall(__syscall_open, filename, 65, 0x1fd);
+#endif
+/* FIXME: mode not work currently in riscv */
+#ifdef __riscv
+    return __syscall(__syscall_openat, -100, filename, 65, 0x1fd);
+#endif
     if (!strcmp(mode, "rb"))
+#ifdef __arm__
         return __syscall(__syscall_open, filename, 0, 0);
+#endif
+#ifdef __riscv
+    return __syscall(__syscall_openat, -100, filename, 0, 0);
```

Comment:
I am not sure if this change is necessary. Can you explain?

Reply:
rv32 doesn't support system call `open` (at least 1024 isn't the system call number for `open`). Also, in this [ref](https://fedora.juszkiewicz.com.pl/syscalls.html) show that rv32 doesn't support `open`.

Reply:
Append the above information within the comments.

Reply:
I would be afraid of the conditional compilation, which looks a bit ugly without `#else` and/or `#elif`.

Reply:
Got it, I will append above information. Also, I'll send a issue about lack of feature `#else` and `#elif`. It seems fine to use only `#ifdef` now. Feature about `#else` and `#elif` will be added in future PR.